Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/TypeResolution/InteractableTypeFinder.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Interactable.TypeResolution
+{
+    /// <summary>
+    /// A helper that uses reflection to find objects that implement base types of the
+    /// Interactable types that populate the various state, theme, and event inspectors.
+    /// </summary>
+    public class InteractableTypeFinder
+    {
+        /// <summary>
+        /// Controls the behavior of the InteractableTypeFinder.FindTypes function. See individual
+        /// enum values for more details.
+        /// </summary>
+        public enum TypeRestriction
```

Comment:
is this enum intended to be referenced by client code? if so, our guidelines ask for a separate code file. If not, please mark as private.

thanks!

Reply:
While it's not going to be referenced by customer client code (i.e. this is an implementation detail of how the MRTK guts looks up these values), it's still located in a location where labelling it private will mean the other internal code won't be able to access it. I've gone ahead with its own file.